Src/AutoFixture/FreezeOnMatchCustomization.cs

```diff
@@ -75,6 +80,41 @@ public FreezeOnMatchCustomization(Type targetType)
         public IRequestSpecification Matcher { get; }
 
         /// <summary>
+        /// Used to store a request for resolving a specimen before freezing.
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// By default the TargetType property is used as the request, but it can be 
+        /// adjusted to return any other type of request if necessary.
+        /// </para>
+        /// <para>
+        /// A typical use case is to define a request which includes significant 
+        /// information about a specimen. For instance, <see cref="ParameterInfo"/> 
+        /// which is widely used in Glue Libraries, can contain custom attributes
+        /// such as <see cref="StringLengthAttribute"/> or <see cref="RegularExpressionAttribute" />.
+        /// These attributes are essential part of the request since they influence 
+        /// the final specimen. In order not to loose this info, the <see cref="ParameterInfo"/>
+        /// should be set into the Request property.
+        /// </para>
+        /// </remarks>
+        public object Request
+        {
+            get
+            {
+                return this.request;
+            }
+            set
```

Comment:
@ecampidoglio you know, there was a long discussion in #317, we discussed it [recently in #665](https://github.com/AutoFixture/AutoFixture/pull/665#discussion_r70165230). Those discussions revealed so many pros and cons that finding an acceptable solution looks almost unreachable.

In order to fix #637, the `TargetType` needs to be changes in a _single place_ hidden from the consumer eyes (deep in `[FrozenAttribute]`). Now I don't think it worth changing `TargetType` to `object` publicly in the `FreezeOnMatchCustomization` (opening [undesired use cases](https://github.com/AutoFixture/AutoFixture/pull/317#discussion_r19219817)) because of this tiny change. Does it may sense?